File: Selenium/Project/Activity7.py
"""Reading additional information 
Goal: Reading a popup that contains additional information about the account/lead. 
a. Open the browser to the Alchemy CRM site and login.
b. Navigate to Sales -> Leads
c. In the table, find the Additional information icon at the end of the row of the lead 
information. Click it. 
d. Read the popup and print the phone number displayed in it. 
e. Close the browser. """

#Import the webdriver
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.wait import WebDriverWait
import time

#Initialize the firefox driver
with webdriver.Firefox() as driver:
    #Open the browser
    driver.get("http://alchemy.hguy.co/crm")

    #Find the username & password. Enter the credentials and click login
    username = driver.find_element(By.NAME, "user_name")
    password = driver.find_element(By.NAME, "username_password")
    username.send_keys("admin")
    password.send_keys("pa$$w0rd")
    driver.find_element(By.NAME, "Login").click()

    #Navigate to sales -> leads
    driver.find_element(By.XPATH, "//a[contains(@id, 'grouptab_0')]").click()
    driver.find_element(By.XPATH, "//a[contains(@id, 'moduleTab_9_Leads')]").click()
    
    #Set the implicit wait
    #driver.implicitly_wait(15)
    
    #Wait for the first element to be visible
    wait = WebDriverWait(driver, 10)

    first_element = wait.until(EC.visibility_of_element_located((By.XPATH, "//span[@class='suitepicon suitepicon-action-info']")))
    first_element.click()
    print("Additional details popped up")
    time.sleep(2)
    
    #Wait for the popup to be visible
    popup = wait.until(EC.visibility_of_element_located((By.XPATH, "/html/body/div[4]/div/div[7]/div[2]/h2")))

    #Find the phone number inside the popup and print it
    phoneNo = driver.find_element(By.XPATH, "//span[@class='phone']")
    print("Phone number is ", phoneNo.text)
    
    #Find the close button and click it to close the popup
    closePopUp = driver.find_element(By.XPATH, "(//span[contains(@class, 'ui-button-icon-primary ui-icon ui-icon-closethick')])[4]")
    closePopUp.click()

    time.sleep(2)
    driver.quit()

    
    
    

File: Selenium/Project/Activity7.java
/*
 1. Open Alchemy CRM browser and login
 2. Navigate to Sales -> leads
 3. In the table, find the additional information icon at the end of the row of the lead information. Click it.
 4. Read the popup and print the phone number displayed in it.
 5. Close the browser.
 */
package crmactivities;

import java.time.Duration;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;
import org.testng.annotations.AfterClass;

public class Activity7 {

	//Driver initialization
	WebDriver driver;
	WebDriverWait wait;
	
	@BeforeClass
	public void setUp() {
		
		driver = new FirefoxDriver();
		driver.get("http://alchemy.hguy.co/crm");
		
		WebElement userName = driver.findElement(By.name("user_name"));
		WebElement password = driver.findElement(By.name("username_password"));
		
		//Enter login credential
		userName.sendKeys("admin");
		password.sendKeys("pa$$w0rd");
		driver.findElement(By.name("Login")).click();
	}
	
	@Test
	public void navigateToSales() throws InterruptedException {
		
		//Navigate to sales
		driver.findElement(By.xpath("//a[contains(@id, 'grouptab_0')]")).click();
		driver.findElement(By.xpath("//a[contains(@id, 'moduleTab_9_Leads')]")).click();
		
		//Implicit wait
		//driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(20));
		
		wait = new WebDriverWait(driver, Duration.ofSeconds(20));
		WebElement firstElement = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//span[@class='suitepicon suitepicon-action-info']")));
		firstElement.click();
		Thread.sleep(2000);
		
		WebElement popup = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("/html/body/div[4]/div/div[7]/div[2]/h2")));
		
		WebElement phoneNo = popup.findElement(By.xpath("//span[@class='phone']"));
		String printPhoneNo = phoneNo.getText();
		System.out.println("Phone number is " + printPhoneNo);
		
		WebElement closePopup = popup.findElement(By.xpath("(//span[contains(@class, 'ui-button-icon-primary ui-icon ui-icon-closethick')])[4]"));
		closePopup.click();
		
	}
	
	@AfterClass
	public void tearDown() {
		driver.close();
	}
}
